src/test/debuginfo/type-names.rs

```diff
@@ -1,208 +1,263 @@
+// cdb-check:struct type_names::GenericStruct<type_names::mod1::Enum2, f64> mut_generic_struct = [...]
+
+// ENUMS
+// cdb-command:dv /t *_enum_*
+// cdb-check:union type_names::Enum1 simple_enum_1 = [...]
+// cdb-check:union type_names::Enum1 simple_enum_2 = [...]
+// cdb-check:type_names::mod1::Enum2 simple_enum_3 = [...]
+// cdb-check:type_names::mod1::mod2::Enum3 generic_enum_1 = [...]
+// cdb-check:type_names::mod1::mod2::Enum3 generic_enum_2 = [...]
+
+// TUPLES
+// cdb-command:dv /t tuple*
+// cdb-check:struct tuple<u32, type_names::Struct1, type_names::mod1::mod2::Enum3<type_names::mod1::Struct2> > tuple1 = [...]
+// cdb-check:struct tuple<tuple<type_names::Struct1, type_names::mod1::mod2::Struct3>, type_names::mod1::Enum2, char> tuple2 = [...]
+
+// BOX
+// cdb-command:dv /t box*
+// cdb-check:struct tuple<alloc::boxed::Box<f32, alloc::alloc::Global>, i32> box1 = [...]
+// cdb-check:struct tuple<alloc::boxed::Box<type_names::mod1::mod2::Enum3<f32>, alloc::alloc::Global>, i32> box2 = [...]
+
+// REFERENCES
+// cdb-command:dv /t *ref*
+// cdb-check:struct tuple<ref type_names::Struct1*, i32> ref1 = [...]
+// cdb-check:struct tuple<ref type_names::GenericStruct<char, type_names::Struct1>*, i32> ref2 = [...]
+// cdb-check:struct tuple<ref mut type_names::Struct1*, i32> mut_ref1 = [...]
+// cdb-check:struct tuple<ref mut type_names::GenericStruct<type_names::mod1::Enum2, f64>*, i32> mut_ref2 = [...]
+
+// RAW POINTERS
+// cdb-command:dv /t *_ptr*
+// cdb-check:struct tuple<mut type_names::Struct1*, isize> mut_ptr1 = [...]
+// cdb-check:struct tuple<mut isize*, isize> mut_ptr2 = [...]
+// cdb-check:struct tuple<mut type_names::mod1::mod2::Enum3<type_names::Struct1>*, isize> mut_ptr3 = [...]
+// cdb-check:struct tuple<const type_names::Struct1*, isize> const_ptr1 = [...]
+// cdb-check:struct tuple<const isize*, isize> const_ptr2 = [...]
+// cdb-check:struct tuple<const type_names::mod1::mod2::Enum3<type_names::Struct1>*, isize> const_ptr3 = [...]
+
+// VECTORS
+// cdb-command:dv /t *vec*
+// cdb-check:struct tuple<array<type_names::Struct1,3>, i16> fixed_size_vec1 = [...]
+// cdb-check:struct tuple<array<usize,3>, i16> fixed_size_vec2 = [...]
+// cdb-check:struct alloc::vec::Vec<usize, alloc::alloc::Global> vec1 = [...]
+// cdb-check:struct alloc::vec::Vec<type_names::mod1::Enum2, alloc::alloc::Global> vec2 = [...]
+// cdb-command:dv /t slice*
+// cdb-check:struct ref slice<usize> slice1 = [...]
+// cdb-check:struct ref slice<type_names::mod1::Enum2> slice2 = [...]
+
+// TRAITS
+// cdb-command:dv /t *_trait
+// cdb-check:struct type_names::ref mut __dyn<Trait2<type_names::mod1::mod2::Struct3, type_names::GenericStruct<usize, isize> > >* generic_mut_ref_trait = [...]
+// cdb-check:struct type_names::ref __dyn<Trait2<type_names::Struct1, type_names::Struct1> >* generic_ref_trait = [...]
+// cdb-check:struct type_names::Box<__dyn<type_names::Trait2<i32, type_names::mod1::Struct2> >, alloc::alloc::Global> generic_box_trait = [...]
```

Comment:
@michaelwoerister I figured out what was happening here with `Box` being placed in the wrong namespace and have a fix in my next commit.